File: src/main/java/io/karma/pda/foundation/component/EntityImage.java
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

package io.karma.pda.foundation.component;

import io.karma.pda.api.app.component.AbstractComponent;
import io.karma.pda.api.app.component.ComponentType;
import io.karma.pda.api.state.MutableState;
import io.karma.pda.api.state.Synchronize;
import net.minecraft.world.entity.EntityType;

import java.util.UUID;

/**
 * @author Alexander Hinze
 * @since 13/04/2024
 */
public class EntityImage extends AbstractComponent {
    @Synchronize
    public final MutableState<EntityType<?>> entity = MutableState.of(EntityType.CREEPER);

    public EntityImage(final ComponentType<?> type, final UUID id) {
        super(type, id);
    }
}

File: src/composable/kotlin/io/karma/pda/composition/component/EntityImage.kt
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

@file:JvmName("ComponentExtensionsKt")
@file:JvmMultifileClass

package io.karma.pda.composition.component

import io.karma.pda.api.app.component.Container
import io.karma.pda.composition.Composable
import io.karma.pda.composition.Composer
import io.karma.pda.composition.LayoutProps
import io.karma.pda.foundation.component.DefaultComponents
import io.karma.pda.foundation.component.EntityImage

/**
 * @author Alexander Hinze
 * @since 07/06/2024
 */

@Composable
inline fun <C : Container> Composer<C>.EntityImage(
    crossinline layoutProps: LayoutProps,
    localName: String? = null,
    props: @Composable EntityImage.() -> Unit = {}
) = component(
    type = DefaultComponents.ENTITY_IMAGE,
    layoutProps = layoutProps,
    localName = localName,
    props = props
)